client/client-multi/client-c/src/kaa/kaa_extension.c

```diff
@@ -0,0 +1,101 @@
+/*
+ *  Copyright 2016 CyberVision, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+#include "kaa_extension.h"
+
+#include <kaa_extension_private.h>
+
+#include <stddef.h>
+
+#define EXTENSION_COUNT (sizeof(kaa_extensions)/sizeof(*kaa_extensions))
+
+/**
+ * A list of kaa contexts.
+ */
+static void *kaa_extension_contexts[EXTENSION_COUNT] = { NULL };
+
+/**
+ * Return the index of extension in the kaa_extensions.
+ */
+static size_t extension_id_to_idx(kaa_extension_id id)
+{
+    for (size_t i = 0; i < EXTENSION_COUNT; ++i) {
+        if (kaa_extensions[i]->id == id) {
+            return i;
+        }
+    }
+    return -1;
```

Comment:
Or just provide macro INVAL_EXTENSION (or something like that) 
with body of (size_t)-1